bot/plotting/comparison_generator.py

```diff
@@ -1,275 +1,186 @@
+        number=None,
+        param_to_vary=None,
+        bounds=None,
+    ):
+        self.models_for_comp = models_for_comp
+        self.chemistry = chemistry
+        self.is_experiment = is_experiment
+        self.cycle = cycle
+        self.number = number
+        self.param_to_vary = param_to_vary
+        self.bounds = bounds
+        self.parameter_values = pybamm.ParameterValues(chemistry=self.chemistry)
+        self.experiment = (
+            dict(list(enumerate([pybamm.Experiment(self.cycle * self.number)])))
+            if self.cycle is not None
+            else None
+        )
+        self.comparison_dict = {}
+
+    def calculate_t_end(self, parameter_values_for_comp):
+        """
+        Calculates the t_end for t_eval (t_eval=[0, t_end])
+        Parameter:
+            parameter_values_for_comp: dict
+                Of the form -
+                {
+                    0: pybamm.ParameterValues,
+                    1: pybamm.ParameterValues
+                }
+        """
+        # find the minimum value for "Current function [A]"
+        min_curr_value = min(
+            [
+                item["Current function [A]"]
+                for k, item in parameter_values_for_comp.items()
+            ]
+        )
+        factor = min_curr_value / self.parameter_values["Current function [A]"]
+        t_end = (1 / factor * 1.1) * 3600
+
+        return t_end
+
+    def model_comparison(self):
+        """
+        Generates a comparison GIF with 2 or more models
+        """
+        params = {}
+        if not self.is_experiment:
+
+            # vary "Current function [A]" and "Ambient temperature [K]"
+            params = parameter_value_generator(
+                self.parameter_values.copy(),
+                {
+                    "Current function [A]": (None, None),
+                    "Ambient temperature [K]": (265, 355),
+                },
+            )
+
+        # convert the list containing parameter values to a
+        # dictionary for pybamm.BatchStudy
+        parameter_values_for_comp = dict(
+            list(
+                enumerate([params if not self.is_experiment else self.parameter_values])
+            )
+        )
+
+        t_end = self.calculate_t_end(parameter_values_for_comp)
+
+        batch_study = pybamm.BatchStudy(
+            models=self.models_for_comp,
+            parameter_values=parameter_values_for_comp,
+            experiments=self.experiment,
+            permutations=True,
+        )
+
+        if self.chemistry == pybamm.parameter_sets.Ai2020 and self.is_experiment:
+            batch_study.solve(calc_esoh=False)
+        else:
+            batch_study.solve([0, t_end])
+
+        create_gif(batch_study)
+
+        self.comparison_dict.update(
+            {"varied_values": [], "params": parameter_values_for_comp}
+        )
+
+    def parameter_comparison(self):
+        """
+        Generates a comparison with a single model, by varying a single parameter
```

Comment:
You could also vary "Ambient temperature [K]" here as well as the `param_to_vary`. You could also vary "Current function [A]" (keeping it the same for all comparisons). E.g. you recently tweeted "Doyle-Fuller-Newman model with Chen2020 parameters varying 'Positive electrode exchange-current density [A.m-2]' for a 1.0 C discharge at 25.0°C" but you could also do "Doyle-Fuller-Newman model with Chen2020 parameters varying 'Positive electrode exchange-current density [A.m-2]' for a 2.0 C discharge at 35.0°C"

Happy to leave that for a separate PR though.

Reply:
Oh yes, that will be interesting. It will be better if I add it in another PR, this PR is already getting a bit big